### PR TITLE
DC-3567 Add PUT `data/{provider}/{dataset}/{id}/` to `DataApiV1Sdk`

### PR DESCRIPTION
**Pull request by oleksii-symon-corva-ai**
### Rationale
Users need convenient way to access PUT `data/{provider}/{dataset}/{id}/` data API endpoint.

### Changes
Added PUT `data/{provider}/{dataset}/{id}/` to `DataApiV1Sdk`.

[JIRA ticket](https://corvaqa.atlassian.net/browse/DC-3567)

#### TODO
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: src/corva/api_adapter.py [hunk 1]
```diff
@@ -158,6 +158,31 @@ def insert(
...
+            document: new document data.
+
+        Raises:
+            httpx.HTTPStatusError: if request was unsuccessful.
+
+        Returns:
+            Updated document.
+        """
+
+        response = self.http.put(url=f"data/{provider}/{dataset}/{id_}/", json=document)
```
Comment (finomayato) [inline src/corva/api_adapter.py:180]: is it technically possible for one company provide company name of another company and replace their document?

Comment (oleksii-symon-corva-ai) [reply src/corva/api_adapter.py:180]: This is regulated by permissions of the company making the request. As far as I know company A can request access to `read` some dataset of company B.

Comment (finomayato) [reply src/corva/api_adapter.py:180]: so, it should be technically possible within the sdk and "permission regulating" part is in place in data-api. Correct?

**Comment by oleksii-symon-corva-ai** [comment]
@ville @corva-alex-s @tsilevych please review

**Comment by oleksii-symon-corva-ai** [comment]
@ville @corva-alex-s @tsilevych @finomayato please review

**Pull request merged**